internal/testcomposer/testcomposer.go

```diff
@@ -152,3 +167,47 @@ func (c *Client) doJSONResponse(req *http.Request, expectStatus int, v interface
 
 	return json.NewDecoder(res.Body).Decode(v)
 }
+
+// CheckFrameworkAvailability checks that the requested is available on the backend
+func (c *Client) CheckFrameworkAvailability(ctx context.Context, frameworkName string) error {
+	url := fmt.Sprintf("%s/v1/testcomposer/framework/%s", c.URL, frameworkName)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return err
+	}
+	req.SetBasicAuth(c.Credentials.Username, c.Credentials.AccessKey)
+	resp, err := c.HTTPClient.Do(req)
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+
+	err = c.checkFrameworkRestrictions(*resp, string(body))
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("unexpected response code:'%d', msg:'%s'", resp.StatusCode, string(body))
+	}
+	return nil
+}
+
+// checkFrameworkRestrictions checks specific cases related to framework availability
+func (c *Client) checkFrameworkRestrictions(resp http.Response, body string) error {
+	if resp.StatusCode == http.StatusForbidden && body == forbiddenPreviewError {
+		log.Error().Msg("Looks like you are not part of the preview. To join the preview, please sign up here: https://info.saucelabs.com/scale-cypress-testing.html")
+		return errors.New("not part of preview")
+	}
+	if resp.StatusCode == http.StatusBadRequest && body == unsupportedFrameworkError {
+		log.Error().Msg("The framework you've selected is invalid")
```

Comment:
Can we add the name of the unsupported framework to this list and then give a list of what frameworks we do support.

`"The framework ${framework} is not supported. Supported frameworks are ${listOfFrameworks}"`

Reply:
Currently we don't have any endpoint providing the list of supported framework.
I agree that naming the framework in the error would be better, I'll update this !